src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1055,10 +1057,34 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMTransaction> const& m)
             {
                 // For now, be paranoid and have each validator
                 // check each transaction, regardless of source
-                flags |= SF_SIGGOOD;
+                checkSignature = false;
             }
         }
 
+        if (checkSignature)
+        {
+            // Check the signature before handing off to the job queue.
+            auto valid = checkValidity(app_.getHashRouter(), *stx,
```

Comment:
Fixed. Per discussion, this was originally a request to avoid the overhead of spinning up another job, but as we discussed, the cost of checking a signature is much higher.